include/buffering_system.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+#include "types.h"
+#include <math.h>
+
+class BufferingSystem {
+public:
+  virtual insert_ret_t insert(const update_t &upd) = 0;
+  virtual bool get_data(data_ret_t& data) = 0;
+  virtual flush_ret_t force_flush() = 0;
+  virtual void set_non_block(bool block) = 0;
+
+  virtual ~BufferingSystem() {};
+
+  static double sketch_size(node_id_t num_nodes) {
+    return 42 * sizeof(node_id_t) * pow(log2(num_nodes), 2); 
```

Comment:
where does this magic number come from?

Reply:
This is our new equation for the size of a sketch.
We have 7 * log_2(n^2) * log_{3/2}(n) buckets per node sketch. Each bucket consists of a node_id_t value and a 2 * node_id_t value. This comes out to 3 node_id_t per bucket.

log_2(n^2) => 2 log_2(n)

2 * 7 * 3 = 42

In total we get 42 * log_2(n^2) * log_{3/2}(n)

Reply:
I suppose this means that equation is slightly wrong. I'll fix that.